satpy/composites/__init__.py

```diff
@@ -1164,17 +1164,40 @@ class SelfSharpenedRGB(RatioSharpenedRGB):
     @staticmethod
     def four_element_average_dask(d):
         """Average every 4 elements (2x2) in a 2D array"""
-        def _mean4(data):
+        def _mean4(data, offset=(0, 0), block_id=None):
             rows, cols = data.shape
-            if rows % 2 != 0 or cols % 2 != 0:
-                raise ValueError("Self-sharpening requires arrays with "
-                                 "shapes divisible by 2")
-            new_shape = (int(rows / 2.), 2, int(cols / 2.), 2)
-            data_mean = np.ma.mean(data.reshape(new_shape), axis=(1, 3))
+            rows2, cols2 = data.shape
+            pad = []
+            # we assume that the chunks except the first ones are aligned
+            if block_id[0] == 0:
+                row_offset = offset[0] % 2
+            else:
+                row_offset = 0
+            if block_id[1] == 0:
+                col_offset = offset[1] % 2
```

Comment:
Doesn't this depend on a chunk size that is divisible by 2? Regardless, why does the areas offset matter for the average? Is it because we could be averaging the wrong pixels matching the lower resolution pixels? Why can we assume that the other chunks are aligned?

Reply:
Any chunks size would work. And yes, if you eg remove the first column of the dataset, you need to make sure that the pixels being average are right. In the original dataset columns 1 and 2, 3 and 4, 5 and 6 etc have to be combined, so if the first column is removed, it has to become columns 2 and 3, 4 and 5, etc.
About other chunks being aligned, it's because we don't know the size of the surrounding blocks, so we need this to be true for the averaging to be correct.

Reply:
Ah I see the block_id is 0 for all upper-left edge chunks. Ok this makes more sense now. So we have to assume the reader or user isn't dumb and chunks the data across low-resolution pixels.